plugins/modules/openshift_build.py

```diff
@@ -0,0 +1,274 @@
+    There may be a delay between requesting the build and the time the build is terminated.
+  - This can also restart a new build when the current is cancelled.
+  - Analogous to C(oc cancel-build) and C(oc start-build).
+
+extends_documentation_fragment:
+  - kubernetes.core.k8s_auth_options
+
+options:
+  state:
+    description:
+      - Determines if a Build should be started or cancelled.
+    choices:
+      - started
+      - cancelled
+    default: started
+    type: str
+  build_name:
+    description:
+    - Specify the name of a build which should be re-run.
+    - Mutually exclusive with parameter C(build_config_name).
+    type: str
+  build_config_name:
+    description:
+    - Specify the name of a build config from which a new build will be run.
+    - Mutually exclusive with parameter C(build_name).
+    type: str
+  namespace:
+    description:
+    - Specify the namespace for the build or the build config.
+    type: str
+    required: True
+  build_args:
+    description:
+    - Specify a list of key-value pair to pass to Docker during the build.
+    type: list
+    elements: dict
+    suboptions:
+      name:
+        description:
+          - docker build argument name
+        type: str
+        required: true
+      value:
+        description:
+          - docker build argument value
+        type: str
+        required: true
+  commit:
+    description:
+    - Specify the source code commit identifier the build should use;
+      requires a build based on a Git repository.
+    type: str
+  env_vars:
+    description:
+    - Specify a list of key-value pair for an environment variable to set for the build container.
+    type: list
+    elements: dict
+    suboptions:
+      name:
+        description:
+          - environment variable name.
+        type: str
+        required: true
+      value:
+        description:
+          - environment variable value
+        type: str
+        required: true
+  incremental:
+    description:
+    - Overrides the incremental setting in a source-strategy build, ignored if not specified.
+    type: bool
+  no_cache:
+    description:
+    - Overrides the noCache setting in a docker-strategy build, ignored if not specified.
+    type: bool
+  wait:
+    description:
+    - When C(state) is set to I(started), specify whether to wait for a build to complete
+      and exit with a non-zero return code if the build fails.
+    - When C(state) is set to I(cancelled), specify whether to wait for a build phase to be Cancelled.
+    default: False
+    type: bool
+  wait_sleep:
+    description:
+    - Number of seconds to sleep between checks.
+    - Ignored if C(wait) is not set.
+    default: 5
+    type: int
+  wait_timeout:
+    description:
+    - How long in seconds to wait for a build to complete.
+    - Ignored if C(wait) is not set.
+    default: 120
+    type: int
+  restart:
```

Comment:
I'm wondering if it might be more streamlined to remove this parameter and add a `restarted` state?

Reply:
It makes sense to me, will do that, thanks.